src/util/worker/mod.rs

```diff
@@ -140,68 +151,141 @@ pub fn dummy_scheduler<T: Display>() -> Scheduler<T> {
     Scheduler::new("dummy scheduler", AtomicUsize::new(0), tx)
 }
 
+#[derive(Copy, Clone)]
+pub struct Builder<S: Into<String>> {
+    name: S,
+    batch_size: usize,
+    tick_duration: Duration,
```

Comment:
`tick_duration` is confusing, does it mean `on_tick` will be called at every `tick_duration`? Or `on_tick` will last for `tick_duration`?